src/ecmult_gen.h

```diff
@@ -10,7 +10,68 @@
 #include "scalar.h"
 #include "group.h"
 
+#ifndef USE_ECMULT_STATIC_PRECOMPUTATION
+#define USE_COMB 1
+#endif
+
+#if USE_COMB
+
+#if defined(EXHAUSTIVE_TEST_ORDER)
+
+  /* We need to control these values for exhaustive tests because
+   * the tables cannot have infinities in them (secp256k1_ge_storage
+   * doesn't support infinities) */
+#  if EXHAUSTIVE_TEST_ORDER > 32
+#    define COMB_BLOCKS 52
+#    define COMB_TEETH 5
+#  elif EXHAUSTIVE_TEST_ORDER > 16
+#    define COMB_BLOCKS 64
+#    define COMB_TEETH 4
+#  elif EXHAUSTIVE_TEST_ORDER > 8
+#    define COMB_BLOCKS 86
+#    define COMB_TEETH 3
+#  elif EXHAUSTIVE_TEST_ORDER > 4
+#    define COMB_BLOCKS 128
+#    define COMB_TEETH 2
+#  else
+#    define COMB_BLOCKS 256
+#    define COMB_TEETH 1
+#  endif
+
+#  define COMB_SPACING 1
+
+#else
+
+  /* COMB_BLOCKS, COMB_TEETH, COMB_SPACING must all be positive and the product of the three (COMB_BITS)
+   * must evaluate to a value in the range [256, 288]. The resulting memory usage for precomputation
+   * will be COMB_POINTS_TOTAL * sizeof(secp256k1_ge_storage). */
+  #define COMB_BLOCKS 4
+  #define COMB_TEETH 5
+  #define COMB_SPACING 13
+
+#endif
+
+/* The remaining COMB_* parameters are derived values, don't modify these. */
+#define COMB_BITS (COMB_BLOCKS * COMB_TEETH * COMB_SPACING)
+#define COMB_GROUPED ((COMB_SPACING == 1) && ((32 % COMB_TEETH) == 0))
```

Comment:
I think we should drop `COMB_GROUPED` because it's impossible for `32 % COMB_TEETH` to be 0. (If it's zero, things obviously won't work...but if it's ≥ 32, then several other things break: `COMB_MASK` overflows; the `bits =` expression in ecmult_gen_impl.h:228 left-shifts too far; I also get this bizarre error about the size of `prec` being negative in ecmult_gen.h:71.)

Reply:
In fact, once `COMB_TEETH` gets much past 10 we start running into trouble in `secp256k1_ecmult_gen_context_build` because we put `COMB_POINTS_TOTAL`-many gejs on the stack.

Given that this is a ridiculous thing to do I don't think we should try to support it, e.g. by using heap allocations rather than stack allocations in gen_context_build. We should just assume it doesn't happen and drop `COMB_GROUPED`.

Reply:
You're right, I misread `32 % COMB_TEETH` (which is 0 for 1,2,4,8) as `COMB_TEETH % 32` (which can't be 0). Let me reassess COMB_GROUPED in that light.

Reply:
https://github.com/bitcoin-core/secp256k1/pull/546/commits/4c8ff9b648a7062133922d2d0c945712e77dd753 adds precompiler guards on the comb constants.